### PR TITLE
New expasy url for scanprosite

### PR DESCRIPTION
**Pull request by pascalschlaepferprivate**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #...
The mirror for prosite was changed by expasy from https://www.expasy.org to https://prosite.expasy.org. I changed this and added a message for the change (how it was before and now.)

### CODE REVIEW

File: NEWS.rst [hunk 1]
```diff
@@ -37,6 +37,7 @@ possible, especially the following contributors:
 - Markus Piotrowski
 - Suyash Gupta
 - Vini Salazar (first contribution)
+- Pascal Schläpfer (first contribution)
```
Comment (MarkusPiotrowski) [inline NEWS.rst:40]: Also in `NEWS.rst` the contributors are sorted by their first name.

File: Bio/ExPASy/ScanProsite.py [hunk 2]
```diff
@@ -28,21 +28,21 @@ def __init__(self):
         self.capped = None
         self.warning = None
 
-
-def scan(seq="", mirror="https://www.expasy.org", output="xml", **keywords):
+### October 28th 2020 it was recognised that the main url of prosite changed from https://www.expasy.org to https://prosite.expasy.org. Thus a change in the mirror was issued from https://www.expasy.org to https://prosite.expasy.org.
```
Comment (MarkusPiotrowski) [inline Bio/ExPASy/ScanProsite.py:31]: I'm not sure if we are enforcing a line-length limit now (I guess not) but would be nice to break this into two lines

Comment (MarkusPiotrowski) [reply Bio/ExPASy/ScanProsite.py:31]: Do you know when the change occurred? Not necessarily important but more informative than the date of recognition.

**Review by MarkusPiotrowski — COMMENTED** [review]
@pascalschlaepferprivate I have left some comments.

There is also a test for the `ExPASy` module in our test suite, which must be run in online mode. Without having looked into it, this issue was probably not caught by it and it would be nice to cover this case.
Do you want to give it a try?

File: Bio/ExPASy/ScanProsite.py [hunk 3]
```diff
@@ -28,21 +28,21 @@ def __init__(self):
         self.capped = None
         self.warning = None
 
-
```
Comment (MarkusPiotrowski) [inline Bio/ExPASy/ScanProsite.py:31]: You need to put this empty line back. `black` is complaining about this.

File: NEWS.rst (hunk 1 above)
Comment (pascalschlaepferprivate) [reply NEWS.rst:40]: ok, sorry, I'm new to this :)

File: Bio/ExPASy/ScanProsite.py (hunk 2 above)
Comment (pascalschlaepferprivate) [reply Bio/ExPASy/ScanProsite.py:31]: No, I can only say that it worked October 10th and then not yesterday.

Comment (MarkusPiotrowski) [reply Bio/ExPASy/ScanProsite.py:31]: That's OK then. It seems that at some time between 2008 and 2012 they changed it...(according to their papers)

Comment (MarkusPiotrowski) [reply Bio/ExPASy/ScanProsite.py:31]: ...but obviously they turned-off the old URL just now.

**Comment by pascalschlaepferprivate** [comment]
Ill look into the expasy test.

**Comment by pascalschlaepferprivate** [comment]
> Ill look into the expasy test.

The expasy test would have catched it, however it was already updated to the new url

**Comment by MarkusPiotrowski** [comment]
I'm not sure, the tests are not explicitly calling `ScanProsite.py` itself, where the url is a default argument. However, I cannot rule out that the tested modules use `ScanProsite` internally.

**Comment by pascalschlaepferprivate** [comment]
Good point,
I can add a test with a sequence and check if the result that comes back is
in the right format, even if the content has changed due to new
information...
Cheers,
Pascal

Am Do., 29. Okt. 2020 um 14:52 Uhr schrieb Markus Piotrowski <
notifications@github.com>:

**Comment by mdehoon** [comment]
Can this be included before the next Biopython release?

**Comment by pascalschlaepferprivate** [comment]
When is it?

**Comment by peterjc** [comment]
@JoaoRodrigues volunteered to do the Biopython 1.79 release, and was aiming for early March - I guess something else came up as he hasn't done it yet:

https://mailman.open-bio.org/pipermail/biopython/2021-February/016859.html

**Review by peterjc — APPROVED** [review]
This looks OK to me, @MarkusPiotrowski was there anything else which needed addressing here?

**Comment by pascalschlaepferprivate** [comment]
To write a decent test, is not really possible.
Since new domains can be created at any time, the result of any query can
be different.
Hence a test that will never produce a false negative is not possible.

**Comment by MarkusPiotrowski** [comment]
I think it's OK to merge, but there's a conflict now with NEWS.rst. @pascalschlaepferprivate Can you do a rebase?

**Pull request merged**